astro/src/content/articles/authentication/bff.mdx

```diff
@@ -0,0 +1,319 @@
+- Any change to an API endpoint or new feature added requires increased collaboration and testing between backend and frontend teams.
+- Authorization is tricky when the API is public and may have free and paid features.
+
+The BFF pattern, shown below, solves these problems by splitting the single API server into one API per app.
+
+![Diagram showing the BFF pattern with separate mobile and web APIs](/img/articles/authentication/bff/bff2.png)
+
+
+Each backend can now be considered part of its corresponding app. Features can be developed independently within each backend-frontend pair, without affecting other apps.
+
+
+## What Is BFF For Authentication?
+
+As the SoundCloud example illustrates, BFFs were proposed for developer ease, not for security (authentication and authorization). Most web apps ten years ago had a server (whether it ran PHP, ASP, or Node.js), and most still do today. Even single-page apps (SPAs) that used AJAX (fetch calls) to dynamically update web pages still updated them by calling a server.
+
+It may seem like "BFF" is just another word for a server — and mostly, it is — but in security-focused designs, the distinction matters. If your app doesn't have a server or stores OAuth access tokens in the browser, the BFF pattern serves a specific security purpose, which this article examines.
+
+But first, let's briefly review web development history to understand how web service design became more complex over time to defend against attacks.
+
+## Authentication In A Simple Web App
+
+The diagram below shows the simplest client–server design used by most small web apps (without OAuth).
+
+![Diagram of a simple client-server design](/img/articles/authentication/bff/bff1.svg)
+
+Here, a user logs in to the server by entering a username and password, and the server marks the user as authenticated by storing a cookie in the browser. The browser sends the cookie to the server with subsequent requests, allowing the user to interact with the site securely.
+
+It doesn't matter if the app serves static HTML pages that submit forms to the server – in the old style of PHP or the modern style of HTMX – or is an SPA written entirely in Vue or React that unfolds into an application in the browser. In both cases, the authentication token is a cookie.
+
+In this design, the backend could technically be considered a BFF, since it serves the frontend directly. But in practice, it’s usually just referred to as the server.
+
+## Why Are Cookies More Secure Than The Web Storage API?
+
+The use of a cookie as the authentication credential in the browser is crucial for security, as opposed to storing the credential in the browser's local or session storage. A cookie can be set to `HttpOnly` and `SameSite=Strict` to prevent JavaScript access and ensure the cookie is only sent to the server that created it.
```

Comment:
`as the authentication credential` -> "as the authentication (or authorization) credential"